vllm/engine/async_llm_engine.py

```diff
@@ -518,22 +524,26 @@ async def run_engine_loop(self):
     async def add_request(
         self,
         request_id: str,
-        prompt: Optional[str],
+        inputs: PromptInputs,
         params: Union[SamplingParams, PoolingParams],
-        prompt_token_ids: Optional[List[int]] = None,
         arrival_time: Optional[float] = None,
         lora_request: Optional[LoRARequest] = None,
-        multi_modal_data: Optional[MultiModalData] = None,
     ) -> AsyncStream:
         if self.log_requests:
```

Comment:
I guess not related to this PR but I feel like the body of this `if` should go into a separate function

Reply:
Actually this bothered me as well since it's otherwise redundant to pass both `prompt` and `prompt_token_ids` at the same time. Perhaps it would be better to move this to OpenAI server in another PR?

Reply:
Sure